api/src/Entity/ChecklistItem.php

```diff
@@ -0,0 +1,193 @@
+<?php
+
+namespace App\Entity;
+
+use ApiPlatform\Doctrine\Orm\Filter\SearchFilter;
+use ApiPlatform\Metadata\ApiFilter;
+use ApiPlatform\Metadata\ApiProperty;
+use ApiPlatform\Metadata\ApiResource;
+use ApiPlatform\Metadata\Delete;
+use ApiPlatform\Metadata\Get;
+use ApiPlatform\Metadata\GetCollection;
+use ApiPlatform\Metadata\Link;
+use ApiPlatform\Metadata\Patch;
+use ApiPlatform\Metadata\Post;
+use App\Entity\ContentNode\ChecklistNode;
+use App\InputFilter;
+use App\Repository\ChecklistItemRepository;
+use App\Util\EntityMap;
+use App\Validator\AssertNoLoop;
+use App\Validator\ChecklistItem\AssertBelongsToChecklist;
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\ORM\Mapping as ORM;
+use Gedmo\Mapping\Annotation as Gedmo;
+use Symfony\Component\Serializer\Annotation\Groups;
+use Symfony\Component\Validator\Constraints as Assert;
+
+/**
+ * A ChecklistItem
+ * A Checklist contains a Tree-Structure of ChecklistItems.
+ */
+#[ApiResource(
+    operations: [
+        new Get(
+            security: 'is_granted("CAMP_COLLABORATOR", object) or is_granted("CAMP_IS_PROTOTYPE", object)'
+        ),
+        new Patch(
+            security: 'is_granted("CAMP_MEMBER", object) or is_granted("CAMP_MANAGER", object)'
+        ),
+        new Delete(
+            security: 'is_granted("CAMP_MEMBER", object) or is_granted("CAMP_MANAGER", object)',
+            validate: true,
+            validationContext: ['groups' => ['delete']]
+        ),
+        new GetCollection(
+            security: 'is_authenticated()'
+        ),
+        new Post(
+            denormalizationContext: ['groups' => ['write', 'create']],
+            securityPostDenormalize: 'is_granted("CAMP_MEMBER", object) or is_granted("CAMP_MANAGER", object)'
+        ),
+        new GetCollection(
+            name: 'BelongsToChecklist_App\Entity\ChecklistItem_get_collection',
+            uriTemplate: self::CHECKLIST_SUBRESOURCE_URI_TEMPLATE,
+            uriVariables: [
+                'checklistId' => new Link(
+                    fromClass: Checklist::class,
+                    toProperty: 'checklist',
+                    security: 'is_granted("CAMP_COLLABORATOR", checklist) or is_granted("CAMP_IS_PROTOTYPE", checklist)'
+                ),
+            ],
+        ),
+    ],
+    denormalizationContext: ['groups' => ['write']],
+    normalizationContext: ['groups' => ['read']],
+    order: ['checklist.id', 'id'],
+)]
+#[ApiFilter(filterClass: SearchFilter::class, properties: ['checklist'])]
+#[ORM\Entity(repositoryClass: ChecklistItemRepository::class)]
```

Comment:
how about a unique constraint of position + parent like in ContentNode
https://github.com/ecamp/ecamp3/blob/cd85a9b71d97a546359a926a148e9816786b8e51/api/src/Entity/ContentNode.php#L51

Reply:
If, then we have to use 3 columns: `checklistid`, `parentid`, `position`.

There can be several `checklistid = null` and `position = 1` (in different checklists).